### PR TITLE
[HIVEMALL-54][SPARK] Add an easy-to-use script for spark-shell

### PR DESCRIPTION
**Pull request by maropu**
## What changes were proposed in this pull request?
This pr added a script to automatically download the latest Spark version, compile Hivemall for the version, and invoke spark-shell with the compiled Hivemall binary.
This pr also included a documentation for `hivemall-on-spark` installation.

## What type of PR is it?
Improvement

### What is the Jira issue?
https://issues.apache.org/jira/browse/HIVEMALL-54

## How was this patch tested?
Manually tested.

### CODE REVIEW

**Comment by myui** [comment]
@amaya382 @Lewuathe Could you confirm that the updated `bin/spark-shell` properly works if your spare time?

**Comment by myui** [comment]
@maropu BTW, better to update `incubator-hivemall-site` once this PR is merged.

**Comment by maropu** [comment]
yea, I'll update just after this merged.

**Comment by amaya382** [comment]
@myui 👌

File: docs/gitbook/spark/getting_started/installation.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+```
+
+> #### Notice
+> If you would like to try Hivemall functions on the latest release of Spark, you just say `bin/spark-shell` in a Hivemall package.
+> This command automatically downloads the latest Spark version, compiles Hivemall for the version, and invokes spark-shell with the compiled Hivemall binary.
+
+Then, you load scripts for Hivemall functions.
+
+```
+scala> :load define-all.spark
```
Comment (Lewuathe) [inline docs/gitbook/spark/getting_started/installation.md]: I found current `define-all.spark` assumes `sqlContext` is imported in advance. But the latest spark-shell imports `sqlContext` under Spark session. So it is necessary to rewrite `define-all.spark` or documentation.
```
scala> :load resources/ddl/define-all.spark
Loading resources/ddl/define-all.spark...
<console>:28: error: not found: value sqlContext
       sqlContext.sql("DROP TEMPORARY FUNCTION IF EXISTS hivemall_version")
       ^
```

I think adding below in doc is sufficient.

```
scala> val sqlContext = spark.sqlContext
```

Comment (maropu) [inline docs/gitbook/spark/getting_started/installation.md]: okay, I added the entry in `val sqlContext = spark.sqlContext` in `define-all.spark`(https://github.com/apache/incubator-hivemall/pull/41/commits/17f2dcf375a1c4c0ad979c80c6591143aefe8a1b).

**Comment by myui** [comment]
@amaya382 Have you confirmed? 
@maropu please merge this PR if he say fine.

**Comment by amaya382** [comment]
Looks mostly good, but I found minor issues not directly related to this PR.
Some declarations in `define-all.spark` are incorrect and duplicated.
e.g. `train_arowh`: https://github.com/maropu/incubator-hivemall/blob/AddScriptForSparkShell/resources/ddl/define-all.spark#L32-L36

**Comment by maropu** [comment]
@amaya382 Can you make prs to fix them?

**Comment by maropu** [comment]
Merged.

**Comment by maropu** [comment]
@amaya382 I made a JIRA ticket: https://issues.apache.org/jira/browse/HIVEMALL-65

**Comment by amaya382** [comment]
@maropu okay, I'll do in a few days

**Comment by myui** [comment]
@maropu Could you update `incubator-hivemall-site`? This feature should be documented.

**Pull request closed without merging**